### PR TITLE
Feat/issue 22 improve schema

### LINKED ISSUE #22: The Schema covers the root elements of the value sets but ignores the structure of the records
image

* Green: Are covered
* Yellow: Are ignored (no rules for them) 

In the current situation any json structure (`object`) is accepted as value for the `valueSetValues` property. 

The structure of the actual values is well defined and the way it is consumed by all the application is already defined. An improved schema that will describe this in more details will prevent errors and ensure the quality.  

### References / Samples 
* Schema: https://github.com/ehn-dcc-development/ehn-dcc-valuesets/blob/release/2.1.0/valueset.json
* Value set: https://github.com/ehn-dcc-development/ehn-dcc-valuesets/blob/release/2.1.0/test-result.json

#### Improve Schema
* https://www.jsonschemavalidator.net/s/4OVqD3Bx

### PR DESCRIPTION
**Pull request by skounis**
### References
- Closes #22

### CODE REVIEW

File: valueset.json [hunk 1]
```diff
@@ -19,7 +19,37 @@
...
+            },
+            "system": {
+              "type": "string"
+            }
+          },
+          "required": [
+            "display",
+            "lang",
+            "active",
+            "version",
```
Comment (skounis) [inline valueset.json:48]: as per https://github.com/ehn-dcc-development/ehn-dcc-valuesets/issues/22#issuecomment-951661634

we should consider the case where the version might be an empty string

Comment (dslmeinte) [reply valueset.json:48]: We definitely should cover that in a test case. I'm pretty sure that an empty string satisfies the schema w.r.t. “'version' must be a string”, though.

**Comment by dslmeinte** [comment]
This PR should target the `release/1.3.1` branch.

**Comment by skounis** [comment]
> This PR should target the `release/1.3.1` branch.
What is the purpose of it. The branch started from the version/branch `2.1.0`/`release/2.1.0` which at the time was the latest. If we merge it into `1.3.x` it will bring all the changes introduced in the version that followed. Do we really need it? 

I would suggest to keep is as a new feature we will introduce in the forthcoming release.

File: valueset.json (hunk 1 above)
Comment (skounis) [reply valueset.json:48]: Totally agree with the test case.

**Comment by dslmeinte** [comment]
Sorry, scratch that, I was looking at the Schema repo, which latest released version is 1.3.0.

**Comment by skounis** [comment]
> Sorry, scratch that, I was looking at the Schema repo, which latest released version is 1.3.0.

Since you mentioned it: For the Values sets the schema was moved into this repository

- https://github.com/ehn-dcc-development/ehn-dcc-valuesets/blob/main/valueset.json

I'm not sure if its the right place (I have no strong arguments) nor if it's name, `valueset.json` is intuitive enough. If you see room for improvements let's open a ticket and handle it.

File: valueset.json [hunk 2]
```diff
@@ -19,7 +19,37 @@
...
+            "active": {
+              "type": "boolean"
+            },
+            "version": {
+              "type": "string"
+            },
+            "system": {
+              "type": "string"
+            }
+          },
```
[the remark this answers is not in the document]
Comment (dslmeinte) [reply valueset.json:43]: It's “interesting” (!) where that `validUntil` field comes from. I don't see it in this repo, also not when running the `source-update/update-rat.sh` script. I do see it coming back from our (=NL) National Backend, so apparently it's also returned (and therefore: added) by the EU DCC Gateway. All values for that field are `null`, though.

[the remark this answers is not in the document]
Comment (skounis) [reply valueset.json:43]: @dslmeinte @peterssond The introduction of the `validUntil` field is described in the following ticket

* https://github.com/ehn-dcc-development/ehn-dcc-valuesets/issues/26

[the remark this answers is not in the document]
Comment (peterssond) [reply valueset.json:43]: @skounis I've noticed that aswell, but I'm confused over its status. Neither the ticket nor the pull requests associated with the ticket are resolved, but the `validUntil` field is already in production. The link shows the valueSet from SE NB: https://dgcg.covidbevis.se/tp/valuesets/covid-19-lab-test-manufacturer-and-name

[the remark this answers is not in the document]
Comment (skounis) [reply valueset.json:43]: @peterssond The Pull Request refers to some improvements in the schema. So far, the schema describes only the parent elements and not their details. The implementation of the `validUnitl` is made in the DCC Gateway. It gets the data for JRC and populates this field accordingly. 

If you have an example where the logic fails and values in the JRC data are lost, please share the example.

**Comment by dslmeinte** [comment]
It seems this change is also in PR #26 : can we close this PR and just look at the other one?

**Pull request closed without merging**